packages/workers-shared/asset-server-worker/src/utils/kv.ts

```diff
@@ -18,7 +18,7 @@ export async function getAssetWithMetadataFromKV(
 		} catch (err) {
 			if (attempts >= retries) {
 				throw new Error(
-					`Requested asset ${assetKey} could not be found in KV namespace.`
+					`Requested asset ${assetKey} could not be fetched from KV namespace.`
```

Comment:
I _think_, (but I haven't actually tested), that `getWithMetadata` would only throw in the case of an error in fetching, rather than a non-existent record?

Reply:
that's a good callout! Docs were a bit scarce on this sort of details. I'll try to test and get some clarity